app/src/main/java/com/amaze/filemanager/asynchronous/services/ftp/FtpService.java

```diff
@@ -366,4 +383,51 @@ private static Boolean callIsWifiApEnabled(@NonNull WifiManager wifiManager) {
         return r;
     }
 
+    private Notification buildNotification(){
+        Notification notification;
+        NotificationCompat.Builder builder;
+
+        Intent notificationIntent = new Intent(this, MainActivity.class);
+        notificationIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        PendingIntent contentIntent = PendingIntent.getActivity(this, 0, notificationIntent, 0);
+
+        CharSequence tickerText = this.getString(R.string.ftp_notif_starting);
+        long when = System.currentTimeMillis();
+
+        builder  = new NotificationCompat.Builder(getApplicationContext(), NotificationConstants.CHANNEL_FTP_ID)
+                .setContentTitle(this.getString(R.string.ftp_notif_starting_title))
+                .setContentText(this.getString(R.string.ftp_notif_starting))
+                .setContentIntent(contentIntent)
+                .setSmallIcon(R.drawable.ic_ftp_light)
+                .setTicker(tickerText)
+                .setWhen(when)
+                .setOngoing(true)
+                .setOnlyAlertOnce(true);
+
+        int stopIcon = android.R.drawable.ic_menu_close_clear_cancel;
+        CharSequence stopText = this.getString(R.string.ftp_notif_stop_server);
+        Intent stopIntent = new Intent(FtpService.ACTION_STOP_FTPSERVER).setPackage(getApplicationContext().getPackageName());
+        PendingIntent stopPendingIntent = PendingIntent.getBroadcast(getApplicationContext(), 0,
+                stopIntent, PendingIntent.FLAG_ONE_SHOT);
+
+        builder.addAction(stopIcon, stopText, stopPendingIntent);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN && Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
```

Comment:
All these if can be replaced by `NotificationConstants.setMetadata(getApplicationContext(), builder, NotificationConstants.TYPE_FTP);`